PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/nucleus/NucleusData.java

```diff
@@ -0,0 +1,311 @@
+
+            if (jailed && jailService.getPlayerJailData(user).isPresent()) {
+                Inmate inmate = jailService.getPlayerJailData(user).get();
+
+                String reason = inmate.getReason();
+                String start = "Unknown";
+                String end;
+                String link = "Unknown";
+
+                if (inmate.getJailer().isPresent()) {
+                    User operatorUser = getUser(inmate.getJailer().get());
+                    if (operatorUser != null) {
+                        String operator = operatorUser.getName();
+                        link = Html.LINK.parse(PlanAPI.getInstance().getPlayerInspectPageLink(operator), operator);
+                    }
+                }
+
+                if (inmate.getCreationInstant().isPresent()) {
+                    start = FormatUtils.formatTimeStampYear(inmate.getCreationInstant().get().toEpochMilli());
+                }
+
+                if (inmate.getRemainingTime().isPresent()) {
+                    end = FormatUtils.formatTimeStampYear(inmate.getRemainingTime().get().plusMillis(System.currentTimeMillis()).toMillis());
+                } else {
+                    end = "Permanent jail sentence.";
+                }
+
+                inspectContainer.addValue("&nbsp;" + getWithIcon("Operator", Icon.called("user").of(Color.YELLOW)), link);
+                inspectContainer.addValue("&nbsp;" + getWithIcon("Date", Icon.called("calendar").of(Color.YELLOW).of(Family.REGULAR)), start);
+                inspectContainer.addValue("&nbsp;" + getWithIcon("Ends", Icon.called("calendar-check").of(Color.YELLOW).of(Family.REGULAR)), end);
+                inspectContainer.addValue("&nbsp;" + getWithIcon("Reason", Icon.called("comment").of(Color.YELLOW).of(Family.REGULAR)), reason);
+                inspectContainer.addValue("&nbsp;" + getWithIcon("Jail", Icon.called("bars").of(Color.YELLOW).of(Family.SOLID)), inmate.getJailName());
+            }
+        }
+
+        if (NucleusAPI.getHomeService().isPresent()) {
+            NucleusHomeService homeService = NucleusAPI.getHomeService().get();
+
+            int homeCount = homeService.getHomeCount(user);
+            int maxHomes = homeService.getMaximumHomes(user);
+
+            inspectContainer.addValue("&nbsp;" + getWithIcon("Homes", Icon.called("home").of(Color.GREEN).of(Family.SOLID)), homeCount + "/" + maxHomes);
+
+            List<Home> homes = homeService.getHomes(user);
+
+            if (!homes.isEmpty()) {
+                TableContainer homesTable = new TableContainer(getWithIcon("Home", Icon.called("home").of(Family.SOLID)));
+                homesTable.setColor("light-green");
+
+                for (Home home : homes) {
+                    homesTable.addRow(home.getName());
+                }
+
+                inspectContainer.addTable("Homes", homesTable);
+            }
+        }
+
+        if (NucleusAPI.getInvulnerabilityService().isPresent()) {
```

Comment:
This if block could be refactored into it's own method
`ifPresent(service -> newMethod(service, inspectContainer))`